### PR TITLE
Product creation experience: add more context about product data

### LINKED ISSUE #33539: Product creation experience: add more context about product data
**Description:** The product data naming is far from being self-explanatory and it can be challenging for users to understand the differences between all the product types. This is one of those cases where it's better to provide documentation to help users understand how to complete their tasks.

**Acceptance criteria:**

- [ ] Add a tooltip with a link to a help article about product types: https://woocommerce.com/document/managing-products/#product-types

### PR DESCRIPTION
**Pull request by octaedro**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds a tooltip to provide more context to the `Product data` area, during product creation.

Screen Capture on 2022-07-15 at 13-04-52

Closes #33539.

// cc @jarekmorawski @pmcpinto 

### How to test the changes in this Pull Request:

1. Checkout this branch.
2. Go to `Products` > `Add New`.
3. Verify an interrogation sign (`?`) icon is visible next to the product type selector, in the `Product data` area.
4. Verify that the text in the tooltip looks correct.

| Product tye | Text |
| --------- | -------- |
| `simple` | Simple – covers the vast majority of any products you may sell. Simple products are shipped and have no options. For example, a book. |
| `grouped` | Grouped – a collection of related products that can be purchased individually and only consist of simple products. For example, a set of six drinking glasses. |
| `external` | External or Affiliate – one that you list and describe on your website but is sold elsewhere. |
| `variable` | Variable – a product with variations, each of which may have a different SKU, price, stock option, etc. For example, a t-shirt available in different colors and/or sizes. |

5. Install [this plugin](https://gist.github.com/octaedro/4230eac7e4395a52767392cc21778bef) (it will include a product type to the selector).
6. Verify that the product type `My product` has been added.
7. Select it and verify the tooltip shows the text `Product types define available product details and attributes, such as downloadable files and variations. They're also used for analytics and inventory management.`

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/client/legacy/css/admin.scss [hunk 1]
```diff
@@ -4770,6 +4770,13 @@ img.help_tip {
 			line-height: 24px;
 		}
 
+		.tips_prodcut_types {
```
Comment (louwie17) [inline plugins/woocommerce/client/legacy/css/admin.scss]: Spelling mistake, have to switch the `c` and `u` in `prodcut`

File: plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-panel.php [hunk 2]
```diff
@@ -21,6 +21,7 @@
 				<?php endforeach; ?>
 				</optgroup>
 			</select>
+			<a href="https://woocommerce.com/document/managing-products/#product-types" class="tips_prodcut_types" target="_blank"><?php echo wc_help_tip( __( 'Click in the icon to learn more about each product type', 'woocommerce' ) ); ?></a>
```
Comment (louwie17) [inline plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-panel.php]: Same spelling issue as here as above in the class name.

**Review by louwie17 — COMMENTED** [review]
This tested well @octaedro, I just noticed one spelling mistake in the class name. 

But as @jarekmorawski mentioned in the [last comment in the original issue](https://github.com/woocommerce/woocommerce/issues/33539#issuecomment-1177903551), the tooltip text does feel a little odd, pointing users to click on it instead, and not something we have done before.
I like the idea that Jarek proposed, or maybe we want to add a link within the tooltip that users can click? (although a recent change I made doesn't allow users to hover over the tooltip content, so that would make clicking a link within a bit difficult, I could update that though).

File: plugins/woocommerce/client/legacy/css/admin.scss (hunk 1 above)
Comment (octaedro) [reply plugins/woocommerce/client/legacy/css/admin.scss]: Nice catch! I fixed it in commit 0188d18

**Comment by octaedro** [comment]
Hey @louwie17,

Could you take another look at this PR?

I added the fix to the typo you mentioned, the changes that [Jarek suggested](https://github.com/woocommerce/woocommerce/issues/33539#issuecomment-1184271711), and updated the testing instructions as well.

File: plugins/woocommerce/client/legacy/css/admin.scss [hunk 3]
```diff
@@ -4769,6 +4769,14 @@ img.help_tip {
 			display: block;
 			line-height: 24px;
 		}
+		.product-data-wrapper {
+			label > * {
+				display: inline-block;
+			}
+			.woocommerce-product-type-tip {
+				margin-left: 1em;
```
Comment (louwie17) [inline plugins/woocommerce/client/legacy/css/admin.scss]: We might want to better reflect how the tip icon looks compared to the other icons:
Screen Shot 2022-07-18 at 4 14 16 PM

They have this css:
```
font-size: 1.4em;
margin-left: 9px;
```

**Review by louwie17 — CHANGES_REQUESTED** [review]
Thanks for the updates @octaedro, nice work on getting the tooltip content to update when the dropdown changes 👍 
I just left one minor suggestion for a styling change to keep in sync with how the other tooltips look. Suggestion in in-line comment.

File: plugins/woocommerce/client/legacy/css/admin.scss (hunk 3 above)
Comment (octaedro) [reply plugins/woocommerce/client/legacy/css/admin.scss]: Good eyes @louwie17! I fixed the icon's style in the commit d4266a3

**Review by louwie17 — APPROVED** [review]
Thanks for sticking with the several changes and iterations of this issue @octaedro, this tested well and code looks good as well, this is ready to ship :shipit: 💯

**Pull request merged**